### PR TITLE
py-pyomo: add v6.7.1

### PR DESCRIPTION
**Pull request by mrmundt**
This PR updates the `py-pyomo` package to reflect the most recent Pyomo release (6.7.1).

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pyomo/package.py [hunk 1]
```diff
@@ -85,6 +90,9 @@ class PyPyomo(PythonPackage):
     depends_on("py-sphinx-rtd-theme@0.5:", when="@6.1:+docs", type=("run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyomo/package.py]: sphinx 5+ and sphinx-rtd-theme 0.6+ are now required

File: var/spack/repos/builtin/packages/py-pyomo/package.py [hunk 2]
```diff
@@ -99,6 +107,7 @@ class PyPyomo(PythonPackage):
     depends_on("py-plotly", when="@6.6:+optional", type=("run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyomo/package.py:108]: All matplotlib versions work (other than 3.6.1)

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Not sure if these changed or these have always been wrong

File: var/spack/repos/builtin/packages/py-pyomo/package.py (hunk 1 above)
Comment (mrmundt) [reply var/spack/repos/builtin/packages/py-pyomo/package.py]: In Spack specifically?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-pyomo/package.py]: According to the `setup.py`

File: var/spack/repos/builtin/packages/py-pyomo/package.py [hunk 3]
```diff
@@ -80,25 +85,29 @@ class PyPyomo(PythonPackage):
     depends_on("py-pybind11", when="@6.1:+tests", type=("run"))
 
     # when docs is requested
-    depends_on("py-sphinx@4:", when="@6.1:+docs", type=("run"))
+    depends_on("py-sphinx@5:", when="@6.1:+docs", type=("run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyomo/package.py]: I'm not actually sure if this is accurate. pyomo 6.1 requires sphinx@3:, not @5:. You should go back through the history and see when it changed. Same for other dependencies.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**